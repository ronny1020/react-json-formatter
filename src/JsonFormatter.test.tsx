import { render } from '@testing-library/react'
import JsonFormatter from '.'

describe('JsonFormatter', () => {
  const json = `{
    "name": "John Doe",
    "age": 30,
    "isStudent": true,
    "hobbies": ["reading", "coding", "gaming"],
    "address": {
      "street": "123 Main St",
      "city": "New York",
      "country": "USA"
    } 
  }`

  it('renders JSON string correctly', () => {
    const { container } = render(<JsonFormatter json={json} />)
    expect(container.textContent).toContain('name')
    expect(container.textContent).toContain('John Doe')
    expect(container.textContent).toContain('age')
    expect(container.textContent).toContain('30')
    expect(container.textContent).toContain('isStudent')
    expect(container.textContent).toContain('true')
    expect(container.textContent).toContain('hobbies')
    expect(container.textContent).toContain('reading')
    expect(container.textContent).toContain('coding')
    expect(container.textContent).toContain('gaming')
    expect(container.textContent).toContain('address')
    expect(container.textContent).toContain('street')
    expect(container.textContent).toContain('123 Main St')
    expect(container.textContent).toContain('city')
    expect(container.textContent).toContain('New York')
    expect(container.textContent).toContain('country')
  })

  it('renders JSON object correctly', () => {
    const { container } = render(<JsonFormatter json={JSON.parse(json)} />)
    expect(container.textContent).toContain('name')
    expect(container.textContent).toContain('John Doe')
    expect(container.textContent).toContain('age')
    expect(container.textContent).toContain('30')
    expect(container.textContent).toContain('isStudent')
    expect(container.textContent).toContain('true')
    expect(container.textContent).toContain('hobbies')
    expect(container.textContent).toContain('reading')
    expect(container.textContent).toContain('coding')
    expect(container.textContent).toContain('gaming')
    expect(container.textContent).toContain('address')
    expect(container.textContent).toContain('street')
    expect(container.textContent).toContain('123 Main St')
    expect(container.textContent).toContain('city')
    expect(container.textContent).toContain('New York')
    expect(container.textContent).toContain('country')
    expect(container.textContent).toContain('USA')
  })

  it('applies custom class names correctly', () => {
    const jsonClassName = {
      braceClassName: 'custom-brace',
      stringClassName: 'custom-string'
    }

    const { container } = render(
      <JsonFormatter json={json} jsonClassName={jsonClassName} />
    )

    const braceElement = container.querySelector('.custom-brace')
    expect(braceElement).toBeInTheDocument()

    const stringElement = container.querySelector('.custom-string')
    expect(stringElement).toBeInTheDocument()
  })

  it('match the snapshot', () => {
    const { container } = render(<JsonFormatter json={json} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
