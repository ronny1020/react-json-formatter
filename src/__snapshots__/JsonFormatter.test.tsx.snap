// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JsonFormatter > match the snapshot 1`] = `
<div
  class=""
>
  <div>
    <span
      class=""
    >
      {
    </span>
    <br />
    <span
      class=""
    >
        
    </span>
    <span
      class=""
    >
      "name": 
    </span>
    <span
      class=""
    >
      "John Doe"
    </span>
    <span
      class=""
    >
      ,
    </span>
    <br />
    <span
      class=""
    >
        
    </span>
    <span
      class=""
    >
      "age": 
    </span>
    <span
      class=""
    >
      30
    </span>
    <span
      class=""
    >
      ,
    </span>
    <br />
    <span
      class=""
    >
        
    </span>
    <span
      class=""
    >
      "isStudent": 
    </span>
    <span
      class=" "
    >
      true
    </span>
    <span
      class=""
    >
      ,
    </span>
    <br />
    <span
      class=""
    >
        
    </span>
    <span
      class=""
    >
      "hobbies": 
    </span>
    <span
      class=""
    >
      [
    </span>
    <br />
    <span
      class=""
    >
          
    </span>
    <span
      class=""
    >
      "reading"
    </span>
    <span
      class=""
    >
      ,
    </span>
    <br />
    <span
      class=""
    >
          
    </span>
    <span
      class=""
    >
      "coding"
    </span>
    <span
      class=""
    >
      ,
    </span>
    <br />
    <span
      class=""
    >
          
    </span>
    <span
      class=""
    >
      "gaming"
    </span>
    <br />
    <span
      class=""
    >
        
    </span>
    <span
      class=""
    >
      ]
    </span>
    <span
      class=""
    >
      ,
    </span>
    <br />
    <span
      class=""
    >
        
    </span>
    <span
      class=""
    >
      "address": 
    </span>
    <span
      class=""
    >
      {
    </span>
    <br />
    <span
      class=""
    >
          
    </span>
    <span
      class=""
    >
      "street": 
    </span>
    <span
      class=""
    >
      "123 Main St"
    </span>
    <span
      class=""
    >
      ,
    </span>
    <br />
    <span
      class=""
    >
          
    </span>
    <span
      class=""
    >
      "city": 
    </span>
    <span
      class=""
    >
      "New York"
    </span>
    <span
      class=""
    >
      ,
    </span>
    <br />
    <span
      class=""
    >
          
    </span>
    <span
      class=""
    >
      "country": 
    </span>
    <span
      class=""
    >
      "USA"
    </span>
    <br />
    <span
      class=""
    >
        
    </span>
    <span
      class=""
    >
      }
    </span>
    <br />
    <span
      class=""
    />
    <span
      class=""
    >
      }
    </span>
  </div>
</div>
`;
